### PR TITLE
Defer heater/fan/stepper turn off if home bump on Z

### PR DESCRIPTION
**Pull request by FanDjango**
As a first step towards #20598, I have changed the logic in `motion.cpp` slightly, so that if a `HOMING_BUMP_MM` for the Z-AXIS is configured, the heaters/fans/steppers will only be turned off for the slow probe approach after the bump.

It also applies to the `DELAY_BEFORE_PROBING`.

This allows them to stay on for a (potentially) long homing approach on Z.

If the homing bump is turned off ( = 0 ) nothing changes - things will be turned off on the first (and only) approach (could be a long approach, of course)

A second step, for a separate PR, would be to allow specifying **which** things to turn off instead of **all**

### CODE REVIEW

File: Marlin/src/module/motion.cpp [hunk 1]
```diff
@@ -1602,25 +1602,26 @@ void homeaxis(const AxisEnum axis) {
       do_homing_move(axis, -ABS(backoff[axis]) * axis_home_dir, homing_feedrate(axis));
   #endif
 
-  do_homing_move(axis, 1.5f * max_length(TERN(DELTA, Z_AXIS, axis)) * axis_home_dir);
-
-  #if BOTH(HOMING_Z_WITH_PROBE, BLTOUCH_SLOW_MODE)
-    if (axis == Z_AXIS) bltouch.stow(); // Intermediate STOW (in LOW SPEED MODE)
```
Comment (Sebazzz) [inline Marlin/src/module/motion.cpp:1620]: Why is this moved?

Comment (FanDjango) [reply Marlin/src/module/motion.cpp:1620]: Need to know if a bump is requested BEFORE calling the first homing move, telling it NEED or NEED_NOT turn off the heaters. 

Actually this was not moved. What was moved is the block
```
  // Determine if a homing bump will be done and the bumps distance
  // When homing Z with probe respect probe clearance
  const bool use_probe_bump = TERN0(HOMING_Z_WITH_PROBE, axis == Z_AXIS && home_bump_mm(Z_AXIS));
  const float bump = axis_home_dir * (
    use_probe_bump ? _MAX(TERN0(HOMING_Z_WITH_PROBE, Z_CLEARANCE_BETWEEN_PROBES), home_bump_mm(Z_AXIS)) : home_bump_mm(axis)
  );
  ```
  But the `diff` shows it slightly otherwise.

File: Marlin/src/module/motion.cpp [hunk 2]
```diff
@@ -1290,7 +1290,7 @@ feedRate_t get_homing_bump_feedrate(const AxisEnum axis) {
 /**
  * Home an individual linear axis
  */
-void do_homing_move(const AxisEnum axis, const float distance, const feedRate_t fr_mm_s=0.0) {
+void do_homing_move(const AxisEnum axis, const float distance, const feedRate_t fr_mm_s=0.0, const bool heaters_off=true) {
```
Comment (sjasonsmith) [inline Marlin/src/module/motion.cpp]: It seems a little strange to have an argument called `heaters_off` which _may or may not_ impact the execution of the function.

If this were named something like `final_move` then it would not misrepresent what is happening, even if the value is ignored.

**Review by sjasonsmith — APPROVED** [review]
This approach seems reasonable and simple enough to me.

File: Marlin/src/module/motion.cpp (hunk 2 above)
Comment (FanDjango) [reply Marlin/src/module/motion.cpp]: I was trained to always leave a tidbit in a submission. By all means, I heartily agree to change that to "final_approach` for example.
Willco ASAP

**Comment by FanDjango** [comment]
** DO NOT ATTEMPT MERGE **

I will need to refactor this, as a previous/recent separate independent commit on exactly these code lines have "pulled the carpet out from under my feet" so to speak. In order to make that transparent, I will rebase/resolve conflicts/squish and then test and invite reviews anew. Please stand by until then.

**Comment by thinkyhead** [comment]
I found the merge pretty straightforward. If you still want to squash down to a single commit, it's easiest to use `git reset`:
```
git fetch origin
git reset --hard origin/heaters_off_later

git fetch upstream
git merge upstream/bugfix-2.0.x

git reset upstream/bugfix-2.0.x

git add .
git commit -m "'Quiet probing' only on last bump"
```

**Comment by FanDjango** [comment]
That's ok like that. I'll give it a spin later today  with and without enabling heaters off to be sure. Printers all busy right now.

**Comment by FanDjango** [comment]
Ok, looking good, tested it and works fine.

**Review by sjasonsmith — APPROVED** [review]

**Comment by Sebazzz** [comment]
In retrospect there is one issue with this PR: the probe is being tared _before_ the heaters are turned off. I should have seen that before but I noticed it only after flashing the new firmware with the Z-offset drifting again.
